source/adapters/cuda/tensor_map.cpp

```diff
@@ -0,0 +1,142 @@
+//
+// Copyright (C) 2024 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda.h>
+#include <ur_api.h>
+
+#include "context.hpp"
+
+struct ur_exp_tensor_map_handle_t_ {
+  CUtensorMap Map;
+};
+
+#define CONVERT(URTYPE, CUTYPE)                                                \
+  if (URTYPE & UrType)                                                         \
+    return CUTYPE;
+
+inline CUtensorMapDataType
+convertUrToCuDataType(ur_exp_tensor_map_data_type_flags_t UrType) {
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_UINT8,
+          CU_TENSOR_MAP_DATA_TYPE_UINT8);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_UINT16,
+          CU_TENSOR_MAP_DATA_TYPE_UINT16);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_UINT32,
+          CU_TENSOR_MAP_DATA_TYPE_UINT32);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_INT32,
+          CU_TENSOR_MAP_DATA_TYPE_INT32);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_UINT64,
+          CU_TENSOR_MAP_DATA_TYPE_UINT64);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_INT64,
+          CU_TENSOR_MAP_DATA_TYPE_INT64);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_FLOAT16,
+          CU_TENSOR_MAP_DATA_TYPE_FLOAT16);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_FLOAT32,
+          CU_TENSOR_MAP_DATA_TYPE_FLOAT32);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_FLOAT64,
+          CU_TENSOR_MAP_DATA_TYPE_FLOAT64);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_BFLOAT16,
+          CU_TENSOR_MAP_DATA_TYPE_BFLOAT16);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_FLOAT32_FTZ,
+          CU_TENSOR_MAP_DATA_TYPE_FLOAT32_FTZ);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_TFLOAT32,
+          CU_TENSOR_MAP_DATA_TYPE_TFLOAT32);
+  CONVERT(UR_EXP_TENSOR_MAP_DATA_TYPE_FLAG_TFLOAT32_FTZ,
+          CU_TENSOR_MAP_DATA_TYPE_TFLOAT32_FTZ);
+  throw "convertUrToCuDataType failed!";
+}
+
+CUtensorMapInterleave
+convertUrToCuInterleave(ur_exp_tensor_map_interleave_flags_t UrType) {
+  CONVERT(UR_EXP_TENSOR_MAP_INTERLEAVE_FLAG_NONE,
+          CU_TENSOR_MAP_INTERLEAVE_NONE);
+  CONVERT(UR_EXP_TENSOR_MAP_INTERLEAVE_FLAG_16B, CU_TENSOR_MAP_INTERLEAVE_16B);
+  CONVERT(UR_EXP_TENSOR_MAP_INTERLEAVE_FLAG_32B, CU_TENSOR_MAP_INTERLEAVE_32B);
+  throw "convertUrToCuInterleave failed!";
+}
+
+CUtensorMapSwizzle
+convertUrToCuSwizzle(ur_exp_tensor_map_swizzle_flags_t UrType) {
+  CONVERT(UR_EXP_TENSOR_MAP_SWIZZLE_FLAG_NONE, CU_TENSOR_MAP_SWIZZLE_NONE);
+  CONVERT(UR_EXP_TENSOR_MAP_SWIZZLE_FLAG_32B, CU_TENSOR_MAP_SWIZZLE_32B);
+  CONVERT(UR_EXP_TENSOR_MAP_SWIZZLE_FLAG_64B, CU_TENSOR_MAP_SWIZZLE_64B);
+  CONVERT(UR_EXP_TENSOR_MAP_SWIZZLE_FLAG_128B, CU_TENSOR_MAP_SWIZZLE_128B);
+  throw "convertUrToCuSwizzle failed!";
+}
+
+CUtensorMapL2promotion
+convertUrToL2promotion(ur_exp_tensor_map_l2_promotion_flags_t UrType) {
```

Comment:
Should this function be named `convertUrToCuL2Promotion` as the exception message implies? It would also match the naming of other methods in this file.

Reply:
Oops yes good catch